renderdoc/driver/gl/wrappers/gl_texture_funcs.cpp

```diff
@@ -1510,13 +1510,24 @@ void WrappedOpenGL::glCopyImageSubData(GLuint srcName, GLenum srcTarget, GLint s
                                             (GLsizei)srcBlockSize[2], srcData.internalFormat)
                     : GetByteSize(srcWidth, srcBlockSize[1], srcBlockSize[2], srcFmt, srcType);
 
-            for(size_t z = 0; z < (size_t)srcDepth; z += srcBlockSize[2])
+            // GetCompressedByteSize() will factor in the 'partial' blocks at image edges when the
+            // image size is not an integer multiple of the block size, so we need to take into
+            // account that in the loop
+            size_t roundedUpDepth =
+                srcIsCompressed
+                    ? ((srcDepth + srcBlockSize[2] - 1) / srcBlockSize[2]) * srcBlockSize[2]
```

Comment:
I think this would be cleaner with the `AlignUp(srcDepth, srcBlockSize[2])` helper function we have.